uvicorn/main.py

```diff
@@ -566,16 +566,18 @@ def run(
         logger.warning("You must pass the application as an import string to enable 'reload' or " "'workers'.")
         sys.exit(1)
 
-    if config.should_reload:
-        sock = config.bind_socket()
-        ChangeReload(config, target=server.run, sockets=[sock]).run()
-    elif config.workers > 1:
-        sock = config.bind_socket()
-        Multiprocess(config, target=server.run, sockets=[sock]).run()
-    else:
-        server.run()
-    if config.uds and os.path.exists(config.uds):
-        os.remove(config.uds)  # pragma: py-win32
+    try:
+        if config.should_reload:
+            sock = config.bind_socket()
+            ChangeReload(config, target=server.run, sockets=[sock]).run()
+        elif config.workers > 1:
+            sock = config.bind_socket()
+            Multiprocess(config, target=server.run, sockets=[sock]).run()
+        else:
+            server.run()
+    finally:
+        if config.uds and os.path.exists(config.uds):
+            os.remove(config.uds)  # pragma: py-win32
```

Comment:
Can we move this to `Server.shutdown`? instead?

Reply:
That seems sensible. I'll give it a try.

Reply:
Thanks. Sorry for the delay on follow up.

Reply:
I've experimented with this but see no way to make it work well. The socket can be owned by *either* `main:run` (for reload and workers) *or* `Server` (otherwise). When moving cleanup to `Server.shutdown` I would have to duplicate it inside the reload and worker code as well.

So `main:run` seems adequate for cleanup, since it owns the `Server` as well.